src/components/select/simple-select/simple-select.test.js

```diff
@@ -0,0 +1,1022 @@
+        "have.attr",
+        "aria-expanded",
+        "true"
+      );
+      selectList().should("be.visible");
+      selectListOptionGroup("1").should("have.text", "Group one");
+    });
+
+    it.each([
+      ["top", "300px", "0px", "200px", "0px"],
+      ["bottom", "0px", "0px", "0px", "0px"],
+      ["left", "200px", "0px", "500px", "0px"],
+      ["right", "200px", "0px", "0px", "500px"],
+    ])(
+      "should render list in %s position when margins are top %s, bottom %s, left %s and right %s",
+      (position, top, bottom, left, right) => {
+        CypressMountWithProviders(
+          <SimpleSelectComponent
+            listPlacement={position}
+            mt={top}
+            mb={bottom}
+            ml={left}
+            mr={right}
+          />
+        );
+
+        selectText().click();
+        commonDataElementInputPreview().should(
+          "have.attr",
+          "aria-expanded",
+          "true"
+        );
+        selectListPosition()
+          .should("have.attr", "data-popper-placement", position)
+          .and("be.visible");
+      }
+    );
+
+    it.each([
+      ["top", "0px", "0px", "0px", "0px"],
+      ["bottom", "600px", "0px", "0px", "0px"],
+      ["left", "200px", "0px", "0px", "900px"],
+      ["right", "200px", "0px", "500px", "0px"],
+    ])(
+      "should flip list to opposite position when there is not enough space to render it in %s position",
+      (position, top, bottom, left, right) => {
+        CypressMountWithProviders(
+          <SimpleSelectComponent
+            listPlacement={position}
+            flipEnabled
+            mt={top}
+            mb={bottom}
+            ml={left}
+            mr={right}
+          />
+        );
+
+        let flipPosition;
+        if (position === "top") {
+          flipPosition = "bottom";
+        }
+        if (position === "bottom") {
+          flipPosition = "top";
+        }
+        if (position === "left") {
+          flipPosition = "right";
+        }
+        if (position === "right") {
+          flipPosition = "left";
+        }
+
+        selectText().click();
+        commonDataElementInputPreview().should(
+          "have.attr",
+          "aria-expanded",
+          "true"
+        );
+        selectListPosition()
+          .should("have.attr", "data-popper-placement", flipPosition)
+          .and("be.visible");
+      }
+    );
+
+    it.each([
+      [true, "not.be.visible"],
+      [false, "be.visible"],
+    ])("should check the disablePortal prop when %s", (state, visibility) => {
+      CypressMountWithProviders(
+        <Dialog open height="60px">
+          <SimpleSelectComponent disablePortal={state} />
+        </Dialog>
+      );
+
+      selectText().click();
+      selectList().should(visibility);
+    });
+  });
+
+  describe("check events for Simple Select component", () => {
+    it("should return onChange event when a list option is selected", () => {
+      const callback = cy.stub();
```

Comment:
put this into beforeEach hook in this describe section like:
```suggestion
      let callback;
    beforeEach(() => {
      callback = cy.stub();
    });
```

and then you can delete it from every test below

Reply:
Done